rrs/tools/common.py

```diff
@@ -0,0 +1,163 @@
+
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+    logger.setLevel(logging.INFO)
+
+    return logger
+
+def get_pv_type(pv):
+    pv_type = ''
+    if '+git' in pv:
+        pv_type = 'git'
+    elif '+svn' in pv:
+        pv_type = 'svn'
+    elif '+hg' in pv:
+        pv_type = 'hg'
+
+    return pv_type
+
+def get_recipe_files(layerdir):
+    from layerindex import recipeparse
+
+    # Exclude lib dir (likely to include templates)
+    exclude_dirs = [os.path.join(layerdir, 'lib') + os.sep]
+    # Exclude sub-layers
+    for root, dirs, files in os.walk(layerdir):
+        for d in dirs:
+            if os.path.exists(os.path.join(root, d, 'conf', 'layer.conf')):
+                exclude_dirs.append(os.path.join(root, d) + os.sep)
+
+    recipe_files = []
+    for root, dirs, files in os.walk(layerdir):
+        if '.git' in dirs:
+            dirs.remove('.git')
+
+        # remove excluded dirs
+        for d in dirs[:]:
+            fullpath = os.path.join(root, d) + os.sep
+            if fullpath in exclude_dirs:
+                dirs.remove(d)
+
+        for f in files:
+            fullpath = os.path.join(root, f)
+            (typename, _, filename) = recipeparse.detect_file_type(fullpath,
+                    layerdir + os.sep)
+            if typename == 'recipe':
+                recipe_files.append(fullpath)
+    return recipe_files
+
+def load_recipes(layerbranch, bitbakepath, fetchdir, settings, logger,
+        recipe_files=None, nocheckout=False):
+    from layerindex import recipeparse
+    from bb.fetch import FetchError
+
+    try:
+        (tinfoil, tempdir) = recipeparse.init_parser(settings,
+                layerbranch.branch, bitbakepath, nocheckout=nocheckout,
+                logger=logger)
+    except recipeparse.RecipeParseError as e:
+        logger.error(str(e))
+        sys.exit(1)
+
+    layer = layerbranch.layer
+    urldir = str(layer.get_fetch_dir())
+    repodir = os.path.join(fetchdir, urldir)
+    layerdir = os.path.join(repodir, str(layerbranch.vcs_subdir))
+
+    d = recipeparse.setup_layer(tinfoil.config_data, fetchdir, layerdir,
+            layer, layerbranch, logger)
+
+    if recipe_files is None:
+        recipe_files = get_recipe_files(layerdir)
+
+    recipes = []
+    for fn in recipe_files:
+        try:
+            logger.debug('Parsing %s' % fn)
+            if hasattr(tinfoil, 'parse_recipe_file'):
+                data = tinfoil.parse_recipe_file(fn, appends=False, config_data=d)
+            else:
+                data = bb.cache.Cache.loadDataFull(str(fn), [], d)
+
+            try:
+                pv = data.getVar('PV', True)
+            except FetchError:
+                data.setVar('SRCPV', '')
+
+            recipes.append(data)
+        except Exception as e:
+            logger.error("%s: branch %s couldn't be parsed, %s" \
+                    % (layerbranch, fn, str(e)))
+            continue
+
+    return (tinfoil, d, recipes, tempdir)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are unpacking more than three variables while using a function that returns more than one variable. Using a large number of return values is prone to errors. We recommend that you return a small class or `namedtuple` instance instead.

[Learn more](https://docs.python.org/3/library/typing.html#typing.NamedTuple)